### PR TITLE
WIP - Add mhv services and pundit policies

### PR DESCRIPTION
**Pull request by saneshark**

### CODE REVIEW

**Comment by saneshark** [comment]
@U-DON this is still a work in progress, but I want to sync up with you and see what we can do in the interim vs longer term. I think presently if we authorize this way accounts will never get created.

File: app/policies/mhvac_policy.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+MHVACPolicy = Struct.new(:user, :mhv_ac) do
+
+  def access?
+    user.loa3? && user.va_patient?
+  end
+
+  def creatable?
+    eligible? &&  user.mhv_account.creatable?
```
[the remark this answers is not in the document]
Comment (saneshark) [reply app/policies/mhvac_policy.rb]: I think the idea here was that `creatable` and `updatable` read better to me than some of the other methods we are using. I was favoring this naming convention mostly for the pundit policy but also eventually for the controller authorization. I will remove for now and we can maybe revisit.

**Comment by saneshark** [comment]
@U-DON lets sync up on this when you're available. I didn't bother to try and fix up the specs yet. Mostly just wanted to get your feedback. I agree with you that separating services from user makes good long term sense. I would love to have an authorizations or services controller, but we'd have to identity all of the places that currently depend on this being available in user. Unless you're proposing having it in both places. I could see a case for it being available in user as a nested attribute perhaps.

Let's try and sync on this soon. Would like to get these changes merged soon.

**Comment by saneshark** [comment]
@U-DON could you give this a second pass?

It touches a lot of files, I also had to remove the MHVAccountCreation error class, couldn't see how it would ever get used anymore. I think there is also some room to cleanup some additional request specs that use mhv_account instead of the authorization policies, but we're already at 20+ files modified so figured I'd push up what I've got for now.

File: spec/request/messages_request_spec.rb [hunk 2]
```diff
@@ -151,15 +150,9 @@
...
-  context 'with a user that does not have access' do
-    let(:mhv_account) do
-      double('mhv_account',
-             accessible?: false,
-             eligible?: false,
-             needs_terms_acceptance?: false,
-             upgraded?: false)
-    end
-    let(:current_user) { build(:user, :loa1) }
+  context 'with a user that does not PREMIUM' do
```
Comment (U-DON) [inline spec/request/messages_request_spec.rb]: Correction: `with a user that is not PREMIUM`?

File: spec/request/messages_request_spec.rb [hunk 3]
```diff
@@ -170,22 +163,21 @@
...
-  context 'with a user that has not accepted T&C' do
-    let(:mhv_account) do
-      double('mhv_account',
-             accessible?: false,
-             eligible?: true,
-             needs_terms_acceptance?: true,
-             upgraded?: false)
+  context 'with a user that is not a va patient', :skip_mvi do
+    let(:mhv_account_type) { 'Premium' }
+    let(:current_user) { build(:user, :mhv, mhv_account_type: mhv_account_type) }
```
Comment (U-DON) [inline spec/request/messages_request_spec.rb]: Doesn't setting `:mhv` already set the `mhv_account_type` to `'Premium'`, or you're just being explicit here?

**Pull request merged**